### ISSUE TITLE
I2C: I2C driver cannot work after waken up from deep sleep @ARC

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#1582]
**_Reported by Qiu Peiyang:_**

Configuration
==========
<Arch> arc
<Board> Quark Se Devboard
<Kernel> unified

Bug detailed description
===================
I2C driver cannot work after resuming from deep sleep and cannot read I2C device.

Testcase to Reproduce
==================
1. Apply PM test framework, https://gerrit.zephyrproject.org/r/#/c/11215/
2. Apply I2C deep sleep case, https://gerrit.zephyrproject.org/r/#/c/11250/
3. Connect I2C GY271 to I2C_0_SS_SCL and I2C_0_SS_SDA.
4. source zephyr-env.sh
5. make BOARD=quark_se_c1000_ss_devboard -C tests/drivers/i2c/i2c_deepsleep flash

Expected Result
=============
I2C driver works well before and after deep sleep. Driver can always read GY271 successfully.
I2C device will be busy for a while, but data transfer will complete soon and return data from GY271.

Actual Result
==========
I2C driver cannot work well after deep sleep. Driver can only read GY271 before deep sleep and fail to read GY271 after resuming and I2C device always keeps busy without returning any data.

Deep sleep entry!
Target device [I2C_0]: to be suspended
I2C device suspend start
system suspend pending on busy I2C_0
Waken up from Deep Sleep
[Deep Sleep]: cycles(437) elapsed from suspend to resume

(Imported from Jira ZEP-1729)

**Comment by nashif** [comment]
**_by Qiu Peiyang:_**

The key words in this case are:
...
I2C device works well before deep sleep.
...
I2C device resume
...
I2C device works well after deep sleep.
...

**Comment by nashif** [comment]
**_by Anselmo L. S. Melo:_**

_Iván Briano_ , _Andre Guedes_ , please take a look.

**Comment by nashif** [comment]
**_by Anselmo L. S. Melo:_**

_Qiu Peiyang_ , please check _Andre Guedes_  comments on GH-1485. This bug looks either related or a duplicate of that one.

**Comment by nashif** [comment]
**_by Anselmo L. S. Melo:_**

_Andre Guedes_ , _Iván Briano_ , confirmed as a dup of GH-1485? Tks.

**Comment by nashif** [comment]
**_by Andre Guedes:_**

Not exactly a duplication of GH-1485 but the solution looks like the same one. _Qiu Peiyang_  could you please confirm if this bug goes away if you run the examples/quark_se/start_arc application from QMSI repository in the x86 core?

**Comment by nashif** [comment]
**_by Anselmo L. S. Melo:_**

_Qiu Peiyang_ , news on this?

**Comment by nashif** [comment]
**_by Anselmo L. S. Melo:_**

_Qiu Peiyang_ , updates?

**Comment by nashif** [comment]
**_by Qiu Peiyang:_**

I tried two start_arc apps, one from Sharron and one compiled by myself, both won't resume from deep sleep.
Please help check whether I use the wrong command.
Commands to compile the start_arc:
make SOC=quark_se TARGET=x86 ENABLE_RESTORE_CONTEXT=1

**Comment by nashif** [comment]
**_by Qiu Peiyang:_**

Verify with new case on both lmt and arc, the issue is gone now.
Looks like the app on x86 is the root cause of the issue.
New case is based on zephyr/tests/power/multicore.
X86 ===== arc
active ==== active
active ==== gpio operations
active ==== idle
idle ==== idle
active ==== active(waken up by x86)
active ==== gpio operations
The new I2C case is attached.
Close this bug.

**Comment by nashif** [comment]
Related to GH-1485

**Comment by nashif** [comment]
**_by Qiu Peiyang:_**

* [test_i2c_deepsleep_arc.tgz](https://s3.amazonaws.com/logs.zephyrproject.org/jira/GH-1582/test_i2c_deepsleep_arc.tgz) (size: 4582)
